discovery/gossiper.go

```diff
@@ -1787,9 +1803,37 @@ func (d *AuthenticatedGossiper) processNetworkAnnouncement(
 		}
 
 		// Before we perform any of the expensive checks below, we'll
-		// make sure that the router doesn't already have a fresher
-		// announcement for this edge.
+		// make sure that we don't process any messages for known zombie
+		// edges.
 		timestamp := time.Unix(int64(msg.Timestamp), 0)
+		if d.cfg.Router.IsZombieEdge(msg.ShortChannelID) {
```

Comment:
similar comment here with unifying `IsZombieEdge` w/ `IsStaleEdgePolicy` to avoid adding another db tx

Reply:
Fixed.